mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -266,10 +267,11 @@ public void init() throws MailboxException {
         if (moveBatcher == null) {
             moveBatcher = new MessageBatcher(MessageBatcher.NO_BATCH_SIZE);
         }
+        MailboxSession session = null;
         if (hasCapability(MailboxCapabilities.Annotation)) {
-            MailboxSession session = null;
             this.addGlobalListener(new MailboxAnnotationListener(mailboxSessionMapperFactory), session);
         }
+        this.addGlobalListener(new MailboxUpdateParentAsLookupListener(mailboxSessionMapperFactory), session);
```

Comment:
addGlobalListener SHOULD be added by the component that requires it. Here JMAP.

The steps to do this are simple:

 - Create in JMAPModule a Configuration perfomer...
 - That takes your mailboxManager AND your mailboxListener...
 - ... and perform at init time the registration...
 - ... then register the configuration performer on the given multibinder...